pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,136 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	BareMetal *BareMetal `json:"bareMetal"` // mandatory for now, possibly optional in future
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking *Networking `json:"networking,omitempty"`
+
+	// Proxy defines the proxy settings for the cluster.
+	// If unset, the cluster will not be configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	UserManagedNetworking  bool   `json:"userManagedNetworking,omitempty"`
```

Comment:
this looks like it could fit into similar discriminator pattern for networking like mentioned here https://github.com/openshift/hive/pull/1247/files#r556741988

Reply:
Property means "use none platform in install config" instead of bare metal. Useful for single-node clusters. 

We think we could infer this by the new provisionRequirements, if control plane count is 1 and workers is 0 or undefined.

Will remove.